job-runner/chain/reaper.go

```diff
@@ -547,9 +547,14 @@ type reaper struct {
 // if sending fails. It returns true if the final state was successfully sent;
 // else false.
 func (r *reaper) sendFinalState(finishedAt time.Time) {
+	fr := proto.FinishRequest{
```

Comment:
Ah, I remember the last bug - need to add RequestId to this struct

Reply:
Thanks! Fixed.